src/modules/hyprland/workspaces.cpp

```diff
@@ -139,15 +172,90 @@ void Workspaces::remove_workspace(std::string name) {
                                 [&](std::unique_ptr<Workspace> &x) { return x->name() == name; });
 
   if (workspace == workspaces_.end()) {
+    // happens when a workspace on another monitor is destroyed
+    return;
+  }
+
+  if ((*workspace)->is_persistent()) {
     return;
```

Comment:
@MonstrousOgre If I just add `(*workspace)->set_windows(0);` here (which gets called when a persistent workspace gets a destroyworkspace event), the whole thing works without listening to the open/close/move-window events. The only problem is the windows_ variable does not show the correct amount: it always shows 0 or 1. Adding more windows does not update the window count.

I can imagine in the future, someone would want to make use of the windows_ variable for something other than knowing whether the workspace is empty or not. With this PR's current implementation, the windows_ variable shows the actual window count.

Reply:
That makes sense